jaxrs-core/src/main/groovy/org/grails/plugins/jaxrs/servlet/http/ReusableStreamHttpServletRequestWrapper.groovy

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2009 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.grails.plugins.jaxrs.servlet.http
+
+import groovy.transform.CompileStatic
+import javax.servlet.ServletInputStream
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletRequestWrapper
+import org.apache.commons.io.IOUtils
+import org.grails.plugins.jaxrs.servlet.ReusableServletInputStream
+
+/** 
+ *
+ * @author Alex Stoia
+ */
+@CompileStatic    
+class ReusableStreamHttpServletRequestWrapper extends HttpServletRequestWrapper {
+    private BufferedInputStream wrappedStream
+    private ServletInputStream usedStream
+        
+    ReusableStreamHttpServletRequestWrapper(HttpServletRequest request) {
+        super(request)
+        ServletInputStream originalStream = request.getInputStream()
+        usedStream = new ReusableServletInputStream(originalStream)
+        usedStream.mark(Integer.MAX_VALUE)
+        // read the stream contents, otherwise this will not be available in 
+        // the getInputStream method. Also tried reading one byte, works on 
+        // first request, fails on all subsequent ones. Got no idea why at
+        // the moment
+        IOUtils.toString(usedStream, 'UTF-8')
```

Comment:
Hi Bud.

I think I found a better solution which involves overwriting the [MultiPartReaderServerSide](https://github.com/jersey/jersey-1.x/blob/b73951e9186fb540675eac62aed622946be781ff/contribs/jersey-multipart/src/main/java/com/sun/jersey/multipart/impl/MultiPartReaderServerSide.java#L65) to support the Grails requests. My question is, you think it is ok if jersey-multipart will be added as a dependency into the grails-jaxrs-jersey1 plugin? I see no reason not to.

Will push the new fix soon